server/src/main/java/com/google/sps/utils/AdRowProcessor.java

```diff
@@ -0,0 +1,117 @@
+        .advertiser(row[ADVERTISER_COLUMN])
+        .startDate(row[START_DATE_COLUMN]) 
+        .endDate(row[END_DATE_COLUMN])
+        .impressionsMin(getImpressionsMin(row[IMPRESSIONS_COLUMN]))
+        .impressionsMax(getImpressionsMax(row[IMPRESSIONS_COLUMN]))
+        .isTargetingAge(getAgeTargets(row[IS_TARGETING_AGE_COLUMN]))
+        .genderTargets(convertStringToList(row[GENDER_TARGETS_COLUMN]))
+        .geoTargets(convertStringToList(row[GEO_TARGETS_COLUMN]))
+        .spendMin(convertStringToLong(row[SPEND_MIN_COLUMN]))
+        .spendMax(convertStringToLong(row[SPEND_MAX_COLUMN]))
+        .headline(row[HEADLINE_COLUMN].trim())
+        .link(row[LINK_COLUMN].substring(3)) // trim "Ad" from "Ad {URL}"
+        .content(row[CONTENT_COLUMN].trim())
+        .headlineSentiment(row[HEADLINE_SENTIMENT_COLUMN].trim())
+        .headlineTerms(row[HEADLINE_TERMS_COLUMN].trim())
+        .contentSentiment(row[CONTENT_SENTIMENT_COLUMN].trim())
+        .contentTerms(row[CONTENT_TERMS_COLUMN].trim())
+        .build();
+    return ad; 
+  }
+
+  /* 
+   * Min and max impression fields contain letters k and M as well as ≤. The 
+   * formatImpressionField helper method makes the fields easier to parse later.
+   */ 
+  public static String[] formatImpressionField(String str) {
+    String impressionsString = str.replace("k", "000").replace("M", "000000").replace(" ","").replace("≤", "");
+    String[] impressionsArray = impressionsString.split("-");
+    return impressionsArray; 
+  }
+
+  public static long getImpressionsMin(String impressionsField) throws IllegalArgumentException {
+    String[] impressionsArray = formatImpressionField(impressionsField); 
+    
+    // If csv field uses "<=", arr has length 1 and min number of impressions is 0.
+    if (impressionsArray.length > 1) {
+      try {
+        return Long.parseLong(impressionsArray[0]);
+      } catch (NumberFormatException e) {
+        System.out.println("Error: " + e);
+        System.exit(0);
+      }
+    }
+    return 0;
+  }
+
+  public static long getImpressionsMax(String impressionsField) throws IllegalArgumentException {
+    String[] impressionsArray = formatImpressionField(impressionsField); 
+    try {
+      return Long.parseLong(impressionsArray[impressionsArray.length - 1]);   
+    } catch (NumberFormatException e) {
+      throw new NumberFormatException();
+    }
+  }
+
+  public static boolean getAgeTargets(String str) {
+    if (str.trim().equals("Not targeted")) {
+      return false; 
+    }
+    return true; 
+  }
+
+  public static List<String> convertStringToList(String str) {
+    List<String> trimmedList = Arrays.stream(str.split(","))
+        .map(String::trim)
+        .collect(Collectors.toList());
+    return trimmedList;  
+  }
+
+  public static long convertStringToLong(String str) {
+    try {
+      return (long) Double.parseDouble(str);
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException();
```

Comment:
This will hide the details/stacktrace from the actual exception `e`. Rethrow the exception, or better still remove the `catch` altogether if it doesn't do anything other than rethrowing. 
Also add `throws IllegalArgumentException` to the method definition.